src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,216 @@
+            unlink(head);
+            found = true;
+            return found;
+        }
+        while (temp != tail && !found) {
+            if (temp.getValue() != null && temp.getValue().equals(object)
+                    || temp.getValue() == object) {
+                unlink(temp);
+                found = true;
+            }
+            if (found) {
+                return found;
+            }
+            temp = temp.getNext();
+        }
+        if (temp.getValue() != null && temp.getValue().equals(object)
+                || temp.getValue() == object) {
+            unlink(tail);
+            found = true;
+        }
+        return found;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return head == null;
+    }
+
+    private ListNode findElementByIndex(int index) {
+        if (index < 0 || index > size - 1 || isEmpty()) {
+            throw new IndexOutOfBoundsException("invalid index " + index);
+        }
+        if (index == 0) {
+            return head;
+        }
+        if (index == size - 1) {
+            return tail;
+        }
+        if (index > size / 2) {
+            ListNode temp = tail;
+            index = size - index - 1;
+            while (index != 0) {
+                temp = temp.getPrev();
+                index--;
+            }
+            return temp;
+        }
+        ListNode temp = head;
+        while (index != 0) {
+            temp = temp.getNext();
+            index--;
+        }
+        return temp;
+    }
+
+    private void unlink(ListNode<T> unlinkElement) {
+        if (unlinkElement == head && size == 1) {
+            head = null;
+            tail = null;
+            size--;
+            return;
+        }
+        if (unlinkElement == tail) {
+            tail.getPrev().setNext(null);
+            tail = tail.getPrev();
+            size--;
+            return;
+        }
+        if (unlinkElement == head) {
+            head = unlinkElement.getNext();
+            size--;
+            return;
+        }
+        unlinkElement.getNext().setPrev(unlinkElement.getPrev());
+        unlinkElement.getPrev().setNext(unlinkElement.getNext());
+        size--;
+    }
+
+    class ListNode<T> {
+        private T value;
+        private ListNode prev;
+        private ListNode next;
+
+        ListNode(T value) {
+            this.value = value;
+        }
+
+        public ListNode getNext() {
+            return next;
+        }
+
+        public ListNode getPrev() {
+            return prev;
+        }
+
+        public T getValue() {
+            return value;
+        }
+
+        public void setNext(ListNode next) {
+            this.next = next;
+        }
+
+        public void setPrev(ListNode prev) {
+            this.prev = prev;
+        }
+
+        public void setValue(T value) {
+            this.value = value;
+        }
```

Comment:
you can use this fields directly

Reply:
@ Noct2000 i am using them directky as i remember but i have to declare methods to pass checks i think. i will lookt at it .